### PR TITLE
feat: gsoc phase 1 report of jaydip

### PR DESCRIPTION
**Pull request by jaydip1235**
Blog of Phase 1

### CODE REVIEW

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        user.update(question_bank_moderator: true)
+      end
+      
+      if users.all? { |user| user.errors.empty? }
+        redirect_back(fallback_location: root_path)
+      else
+        render json: { errors: users.map(&:errors).flat_map(&:full_messages) }, status: :unprocessable_entity
+      end
+    end
+    ```
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:164]: Avoid adding code snippets to this blog, until unless it's too important.
ig All sort of these things are present in your personal blog. Adding link to that blog or code snippet will be better way.

Comment (tanmoysrt) [reply content/posts/Jaydip_Dey_Phase_1_Report.md:164]: Same for all the code snippets in the blog @jaydip1235

Comment (tanmoysrt) [reply content/posts/Jaydip_Dey_Phase_1_Report.md:164]: You may add gif or video of the feature in the blog also to showcase a ppreview of the feature.

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+![phase-1](/images/Jaydip_GSoC24/Phase-1.png)
+
+## About the Project
+
+The project aims to address the need for a comprehensive practice section within CircuitVerse, enhancing learning and engagement in digital logic design. It includes developing a Question Bank Management system for convenient access to categorized questions, each featuring a Circuit Template with input/output probes and a pre-configured testbench. Markdown support will be integrated for improved readability and flexibility in displaying questions. A Question Bank Moderator role will be introduced for selected users to add and modify questions. Answers submitted by users will be automatically verified using CircuitVerse's testbench feature, enhancing the learning experience. Additionally, a Progress Dashboard within user profiles will track submission history and progress, with users able to choose between public or private visibility.
+
+In the first phase, we have covered
+* Community Bonding Period
+* Creating controllers, schemas and tables for all the use case [Week 1 to 3]
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:21]:
```suggestion
* Creating controllers, schemas and tables for circuitverse practice section [Week 1 to 3]
```

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+* Manage question moderators[Week 4]
+* Moderator Question Creation and Integration[Week 5 to 6]
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:23]: Integration with Simulator ?

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+---
+
+### Community Bonding Period
+
+During the community bonding period for GSoC at CircuitVerse, we engaged in various activities to prepare for the project. We began by participating in meet-and-greet sessions with mentors, contributors, and admins, where we shared our excitement for GSoC, discussed our journeys, and introduced our projects. Mentors [Vaibhav Upreti](https://www.linkedin.com/in/vaibhav-upreti/), [Vedant Jain](https://www.linkedin.com/in/vedant-jain-781006145/), and [Tanmoy Sarkar](https://www.linkedin.com/in/tanmoysrt/), answered questions about the CircuitVerse Practice Section project, and [Aboobacker MK](https://www.linkedin.com/in/aboobacker-m-k/) provided insights into the 12-week project process and discussed open defects in CircuitVerse.
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:29]:
```suggestion
During the community bonding period for GSoC at CircuitVerse, there were various activities to provide more context for the project. The project begins with a meet-and-greet sessions with mentors, contributors, and admins, where everyone shared their excitement for GSoC, discussed their journeys, and discuessed about the projects. Mentors [Vaibhav Upreti](https://www.linkedin.com/in/vaibhav-upreti/), [Vedant Jain](https://www.linkedin.com/in/vedant-jain-781006145/), and [Tanmoy Sarkar](https://www.linkedin.com/in/tanmoysrt/), answered questions about the CircuitVerse Practice Section project, and [Aboobacker MK](https://www.linkedin.com/in/aboobacker-m-k/) provided insights into the 12-week project process and discussed open defects in CircuitVerse.
```

You can add a context about `open defects in CircuitVerse`

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+
+Following this, we revisited key technologies by refreshing knowledge of Ruby on Rails through The Odin Project documentation and Vue JS with a YouTube crash course. This preparation was essential for upcoming work on the CircuitVerse codebase.
+
+We then focused on a detailed exploration of the CircuitVerse codebase, identifying areas for modification and making initial changes. we had in-depth discussions with [Ruturaj Mohite](https://www.linkedin.com/in/ruturaj-mohite/), about potential edge cases and the development of the question practice simulator versus the standard simulator, including how to track and save user progress.
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:33]:
```suggestion
Then, the focus got shifted to detailed exploration of the CircuitVerse codebase, identifying areas for modification and making initial changes. we had in-depth discussions with [Ruturaj Mohite](https://www.linkedin.com/in/ruturaj-mohite/), about potential edge cases and the development of the question practice simulator versus the standard simulator, including how to track and save user progress.
```

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+{{< video src="/videos/Jaydip_GSoC24/Add-question.mov" type="video/mp4" preload="auto" >}}
+
+For more detailed information, please refer to [this](https://medium.com/@jaydipdey2807/week-5-6-at-circuitverse-google-summer-of-code-2024-87ea6f738cb4) blog post
+
+---
+
+
+### Pull Request
+
+Pull request for phase 1 can be found [here](https://github.com/CircuitVerse/CircuitVerse/pull/5015)
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:341]: This PR https://github.com/CircuitVerse/CircuitVerse/pull/4967 should be added in the list also.

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+
+
+---
+
+### Future Work
+
+The tasks to be done include developing the UI for the moderators dashboard and integrating it with the backend to enable CRUD operations on questions. Next, efforts will focus on making the question practice page responsive across all screens, implementing filtering logic and backend integration, and setting up auto-verification and submission of questions. Additionally, a progressive dashboard for users will be created to display submission history and progress, followed by backend integration and addition of test cases.
```
Comment (tanmoysrt) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:348]: It will be better if you can write this things in bullet points.

**Review by tanmoysrt — CHANGES_REQUESTED** [review]

**Comment by jaydip1235** [comment]
@tanmoysrt made the changes. Please have a look

File: content/posts/Jaydip_Dey_Phase_1_Report.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+For more detailed information, please refer to [this](https://medium.com/@jaydipdey2807/week-5-6-at-circuitverse-google-summer-of-code-2024-87ea6f738cb4) blog post
+
+---
+
+
+### Pull Request
+
+- PR : [Creating controllers, schemas and tables for circuitverse practice section](https://github.com/CircuitVerse/CircuitVerse/pull/4967)
+- PR : [Addig moderators and questions with simulator integration](https://github.com/CircuitVerse/CircuitVerse/pull/5015)
```
Comment (VaibhavUpreti) [inline content/posts/Jaydip_Dey_Phase_1_Report.md:218]:
```suggestion
- PR : [Adding moderators and questions with simulator integration](https://github.com/CircuitVerse/CircuitVerse/pull/5015)
```

**Comment by jaydip1235** [comment]
@VaibhavUpreti corrected the spelling

**Review by VaibhavUpreti — APPROVED** [review]

**Pull request merged**